### ISSUE TITLE
[v57.0.0] Json import warning on Node 18.20

### ISSUE THREAD

**Issue by kuoruan** [sindresorhus/eslint-plugin-unicorn#2561]
Warnings showing after upgrade to v57.0.0

```js
NODE_OPTIONS='--trace-warnings' eslint .

(node:159194) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
    at emitExperimentalWarning (node:internal/util:274:11)
    at ModuleLoader.jsonStrategy (node:internal/modules/esm/translators:308:3)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:299:14)
```

Caused by:

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/c44ea2641b55a6f244f648010b372db359ff4707/index.js#L4

Can we use this?

```js
import { createRequire } from "module";
const packageJson = createRequire(import.meta.url)("./package.json");
```

**Comment by RobinTail** [comment]
It also causes fatal error in Node 20.9.0.

```
Oops! Something went wrong! :(

ESLint: 9.20.1

SyntaxError: Unexpected token 'with'
    at ModuleLoader.moduleStrategy (node:internal/modules/esm/translators:118:18)
    at callTranslator (node:internal/modules/esm/loader:273:14)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:278:30)
error Command failed with exit code 2.
```

**Comment by RobinTail** [comment]
The same syntax is present in `builtin-modules` that v57 uses, so fixing in one place won't be enough

```js
import builtinModules from './builtin-modules.json' with {type: 'json'};
```

**Comment by fregante** [comment]
Indeed. Import attributes are available on:

v21.0.0, v20.10.0, v18.20.0 (https://nodejs.org/api/esm.html#import-attributes)

But unicorn now has 18.18 in its engines configuration.

**Comment by PrzemyslawMikos** [comment]
Facing the same issue on Node 20.18.0 and ESlint 9.20.1

**Comment by RobinTail** [comment]
I suggest to adjust `engine` in this case because:
- it was a major update (57.0.0) according to SemVer
- the problematic syntax is also present in a subdependency (`builtin-modules`)
- it's easier

According to the [information](https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2561#issuecomment-2676673692) given by @fregante , `engines` could be 

```json
{ "node": "^18.20.0 || ^20.10.0 || >=21.0.0" }
```

**Comment by fisker** [comment]
We won't change how the package.json is imported , see https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2579#issuecomment-2692325274

**Comment by fregante** [comment]
You don't have to change it, the issue is valid and it's about an invalid "engines.node" setup. 

The current value suggests that eslint-plugin-node supports Node 19: that's wrong. 

The solution is to update the `engines` field with accurate information. If the field is not accurate, why is it there at all?

**Comment by Zamiell** [comment]
@fisker Even on Node 22 (the latest LTS at the time of this writing), we still get the warning shown in the original post. Can you clarify that its intended for everyone to get the warning until the feature is no longer experimental? (When does the feature no longer become experimental? In Node v24 maybe? So we have to wait 2 more years?)

**Comment by fregante** [comment]
> Even on Node 22

Update to the latest _Node 22_ version.

No warnings appear in the latest version of each supported Node: v18.20.7 nor v20.18.3 or v22.14.0

I'm updating the `engines` to match the real compatibility in #2581

**Comment by Zamiell** [comment]
> Update to the latest Node 22 version.

Ahh, thank you fregante! That fixed it. This was unintuitive because I did not suspect that they would make such a change like this in a non-breaking version.

> I'm updating the engines to match the real compatibility in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2581

Thank you, that should fix this problem for future users like me! <3
